graphium/data/datamodule.py

```diff
@@ -1160,13 +1170,9 @@ def has_atoms_after_h_removal(smiles):
             self.single_task_datasets, self.task_train_indices, self.task_val_indices, self.task_test_indices
         )
 
-        if self.load_from_file:
+        if self.processed_graph_data_path is not None:
             self._save_data_to_files()
```

Comment:
@DomInvivo not sure this is relevant, but we always cache data if `processed_graph_data_path` is provided.

Are there situations, one wants to read from disk during running the pipeline (e.g., if model doesn't fit in ram otherwise) but discard the data afterwards to not block disk space?

Reply:
If you read from disk, then the data is already there. Deleting it would mean to delete the data from a previous run

Reply:
We need to check if the data is already cached. Otherwise, [this funciton](https://github.com/datamol-io/graphium/blob/7fa23b6fcc59dd07c8f2459706d432e97f6d0f21/graphium/data/datamodule.py#L1435-L1436) will throw an error.